src/box/memtx_tree.cc

```diff
@@ -618,8 +630,7 @@ tree_iterator_start(struct iterator *iterator, struct tuple **ret)
 	if (!res)
 		return 0;
 	*ret = res->tuple;
-	tuple_ref(*ret);
-	it->current = *res;
+	tree_iterator_set_current(it, res);
```

Comment:
May be add assertion that it->current.tuple is NULL here

Reply:
I don't see much point - if it points to a valid tuple, then everything's going to work fine anyway. If it's uninitialized, we'll crash (or this will get caught by asan). There wasn't an assertion here before so I don't think we should add a new one now just for the sake of it.